gossip/c_block_callbacks.go

```diff
@@ -233,11 +237,25 @@ func consensusCallbackBeginBlockFn(
 					}
 
 					block, blockEvents := spillBlockEvents(store, block, es.Rules)
-					txs := make(types.Transactions, 0, blockEvents.Len()*10)
+					unorderedTxs := make([]ScramblerEntry, 0, blockEvents.Len()*10)
+
+					signer := gsignercache.Wrap(types.MakeSigner(chainCfg, blockNumber, uint64(blockCtx.Time)))
 					for _, e := range blockEvents {
-						txs = append(txs, e.Txs()...)
+						for _, tx := range e.Txs() {
+							entry, err := newScramblerTransaction(signer, tx)
+							if err != nil {
+								log.Crit(fmt.Sprintf("cannot create scrambler tx %s", tx.Hash()), "err", err)
+							}
+							unorderedTxs = append(unorderedTxs, entry)
+						}
 					}
 
+					orderedTxs := getExecutionOrder(unorderedTxs, es.Rules.Upgrades.Sonic)
+					txs := make(types.Transactions, len(orderedTxs))
+					for i, tx := range orderedTxs {
+						// Cast back the transactions to pass it to the processor
+						txs[i] = tx.(*scramblerTransaction).Transaction
+					}
```

Comment:
Ideally, this block is outlined into an helper function in `tx_scrambler.go` with a signature like this:
```
func orderTransactions(transactions types.Transactions, rules opera.Rules) types.Transactions
```
This will hide the implementation details of the `ScramblerEntry`, which could help simplify code and makes this function more readable.

Reply:
I wanted to avoid adding another loop, but I agree it will look better.